openssl/src/ssl/mod.rs

```diff
@@ -3114,6 +3112,142 @@ impl SslRef {
         }
         Ok(())
     }
+    #[cfg(not(boringssl))]
+    pub fn set_method(&mut self, method: SslMethod) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::SSL_set_ssl_method(self.as_ptr(), method.as_ptr()))?;
+        };
+        Ok(())
+    }
+    #[corresponds(SSL_use_Private_Key_file)]
+    pub fn set_private_key_file<P: AsRef<Path>>(
+        &mut self,
+        path: P,
+        ssl_file_type: SslFiletype,
+    ) -> Result<(), ErrorStack> {
+        let p = path.as_ref().as_os_str().to_str().unwrap();
+        let key_file = CString::new(p).unwrap();
+        unsafe {
+            cvt(ffi::SSL_use_PrivateKey_file(
+                self.as_ptr(),
+                key_file.as_ptr(),
+                ssl_file_type.as_raw(),
+            ))?;
+        };
+        Ok(())
+    }
+    #[corresponds(SSL_use_PrivateKey)]
+    pub fn use_private_key_pem(&mut self, key: &[u8]) -> Result<(), ErrorStack> {
```

Comment:
Why does this function take a `&[u8]` rather than a `&PKeyRef` (and be called `set_private_key`)? That would be strictly more general.

Reply:
I have change &[u8] to PKey<Private>, but I don't think we should pass as &PKeyRef